scripts/update_jobs.py

```
@@ -656,15 +679,18 @@ def fetch_greenhouse_jobs(company_name: str, url: str, max_retries: int = 2, tim
                 continue
             else:
                 print(f"  ❌ {company_name} request timed out after {max_retries + 1} attempts")
-        except requests.exceptions.RequestException as e:
-            if "404" in str(e):
-                print(f"  ⚠️  {company_name} endpoint not found (404) - company may have moved to a different job board")
-                break  # Don't retry 404s
-            elif attempt < max_retries:
+
+        except requests.exceptions.HTTPError as e:
+            if hasattr(e, 'response') and e.response is not None:
+                if not is_retryable_status(e.response.status_code):
```

Ruff (0.15.6)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)